### PR TITLE
[fix] fix iceberg's filter expr to filter file

### PR DESCRIPTION
**Pull request by wuwenchi**
## Proposed changes

Fix iceberg's filter expr to filter file, and add counts the number of partitions read

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by wuwenchi** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 46.15 seconds
 stream load tsv:          510 seconds loaded 74807831229 Bytes, about 139 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          66 seconds loaded 1101869774 Bytes, about 15 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.2 seconds inserted 10000000 Rows, about 342K ops/s
 storage size: 17162219334 Bytes

**Review by wsjz — APPROVED** [review]
LGTM

**Review by morningman — COMMENTED** [review]
Please add regression test

File: fe/fe-core/src/main/java/org/apache/doris/external/iceberg/util/IcebergUtils.java [hunk 1]
```diff
@@ -372,23 +375,23 @@ private static Object extractDorisLiteral(Expr expr) {
...
             try {
                 date = Date.from(
-                        LocalDateTime.parse(sb.toString(), formatter).atZone(ZoneId.systemDefault()).toInstant());
+                        LocalDateTime.parse(formatDate, dateFormatter).atZone(ZoneId.systemDefault()).toInstant());
             } catch (DateTimeParseException e) {
+                LOG.error("Failed to parse date from \"" + formatDate + "\".");
                 return null;
             }
-            return date.getTime();
+            return date.getTime() * MILLIS_TO_NANO_TIME;
```
Comment (liuchunhua) [inline fe/fe-core/src/main/java/org/apache/doris/external/iceberg/util/IcebergUtils.java:394]: DateLiteral has 'unixTimestamp' method

Comment (wuwenchi) [reply fe/fe-core/src/main/java/org/apache/doris/external/iceberg/util/IcebergUtils.java]: Fixed.

**Comment by wuwenchi** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.94 seconds
 stream load tsv:          508 seconds loaded 74807831229 Bytes, about 140 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          66 seconds loaded 1101869774 Bytes, about 15 MB/s
 stream load parquet:          32 seconds loaded 861443392 Bytes, about 25 MB/s
 insert into select:          29.2 seconds inserted 10000000 Rows, about 342K ops/s
 storage size: 17162219311 Bytes

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**